client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -29,10 +31,25 @@ const DomainRegistrationSuggestion = React.createClass( {
 			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
 			buttonClasses,
 			buttonContent,
-			domainFlag = null;
+			domainFlags = [],
+			translate = this.props.translate;
 
 		if ( suggestion.domain_name ) {
-			domainFlag = <DomainSuggestionFlag domain={ suggestion.domain_name }/>;
+			const newTLDs = [];
+
+			if ( newTLDs.some(
+					( tld ) => endsWith( suggestion.domain_name, tld ) && suggestion.domain_name.substring( 0, suggestion.domain_name.length - ( tld.length + 1 ) ).indexOf( '.' ) === -1
+				) ) {
+				domainFlags.push( <DomainSuggestionFlag content={ translate( 'New', { context: 'Domain suggestion flag' } ) }/> );
+			}
+		}
+
+		if ( suggestion.isRecommended ) {
+			domainFlags.push( <DomainSuggestionFlag content={ translate( 'Recommended', { context: 'Domain suggestion flag' } ) }/> );
```

Comment:
> What is "Recommended" or "New"? A domain name? then the context should be just "domain name". But if the label is not that specific, and it refers to something a bit abstract ("a suggestion"), I'd just drop the context (for all the flags).

Yeah, it refers to a suggestion for a domain name - you know, a "Recommended" suggestion, or a "Best Alternative" one.
I saw the suggestion from @a8ci18n containing `{ context: 'Plugin Installer'}` and thought that the context was probably not needed in that case - and in this one too (and, as a result, the translation could have been shared between those components).